packages/relay/src/lib/eth.ts

```diff
@@ -1017,8 +1019,24 @@ export class EthImpl implements Eth {
         }
       }
 
+      let data = call.data;
+      if (data) {
+        data = crypto.createHash('sha1').update(call.data).digest('hex'); // NOSONAR
+      }
+
+      const cacheKey = `eth_call:.${call.to}.${data}`;
```

Comment:
what happens if `data` is null/unassigned?

Reply:
No problems there we continue with the query as normal, I just didn’t want to calculate hash for a undefined, now that I think, I should’ve added a or call.data.length is bigger than 64(for example).